elements/rh-dialog/rh-dialog.ts

```diff
@@ -80,93 +84,107 @@ export class RhDialog extends LitElement {
    */
   @property({ reflect: true }) position?: 'top';
 
+  /**
+   * Use `accessible-label="My custom label"` to specify the dialog's accessible name.
+   * Defaults to the name of the dialog trigger if no attribute is set and no headings exist in the modal.
+   * See Dialog's Accessibility page for more info.
+   */
+  @property({ attribute: 'accessible-label' }) accessibleLabel?: string;
+
+  /**
+   * `open` / `open="open"` declaratively opens the dialog
+   */
   @property({ type: Boolean, reflect: true }) open = false;
 
   /** Optional ID of the trigger element */
   @property() trigger?: string;
 
+  /** Use `type="video"` to embed a video player into a dialog. */
   @property({ reflect: true }) type?: 'video';
 
   /** @see https://developer.mozilla.org/en-US/docs/Web/API/HTMLDialogElement/returnValue */
   public returnValue = '';
 
   #screenSize = new ScreenSizeController(this);
 
-  @query('#overlay') private overlay?: HTMLElement | null;
-  @query('#dialog') private dialog?: HTMLElement | null;
-  @query('#close-button') private closeButton?: HTMLElement | null;
+  @query('#dialog') private dialog!: HTMLDialogElement;
+  @query('#content') private content!: HTMLElement;
+  @query('#close-button') private closeButton!: HTMLElement;
 
   #headerId = getRandomId();
   #triggerElement: HTMLElement | null = null;
   #header: HTMLElement | null = null;
   #body: Element[] = [];
   #headings: Element[] = [];
   #cancelling = false;
+  #lastTabbable: HTMLElement = this.closeButton;
 
   #slots = new SlotController(this, null, 'header', 'description', 'footer');
 
   connectedCallback() {
     super.connectedCallback();
-    this.addEventListener('keydown', this.onKeydown);
+    this.addEventListener('keydown', this.#onKeyDown);
     this.addEventListener('click', this.onClick);
   }
 
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListener('keydown', this.#onKeyDown);
```

Comment:
```suggestion
```
This should get cleaned up when the element is removed, since the listener is on the host.